### ISSUE TITLE
ServerError (UnsupportedOperationException) in SentinelAPI.download_all()

### ISSUE THREAD

**Issue by arthur-e** [multiply-org/sar-pre-processing#13]
I'm getting a `ServerError` when trying to download files from Sentinel SciHub using the `SentinelAPI`.

I changed the date range so as to download fewer files:

```
Following products will be downloaded
['S1B_IW_SLC__1SDV_20210603T165832_20210603T165900_027195_033F94_5E37'
 'S1B_IW_SLC__1SDV_20210603T165857_20210603T165924_027195_033F94_E158'
 'S1A_IW_SLC__1SDV_20210602T170732_20210602T170759_038164_048116_EF11'
 'S1A_IW_SLC__1SDV_20210601T051818_20210601T051846_038142_048071_F417']
These 4 product need 31.2 Gb of disk space
```

And I set the download path as:

```
import os
path = os.path.join(PROJECT_ROOT, 'download')
try:
    os.makedirs(path)
except: FileExistsError
api.download_all(products, path)
```

Where `PROJECT_ROOT` is defined at the top of my notebook. I see a progress bar for the download but immediately get this error and traceback:

```
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
/usr/local/dev/sar-pre-processing/env/lib/python3.9/site-packages/sentinelsat/sentinel.py in _check_scihub_response(response, test_json, query_string)
   1432     try:
-> 1433         response.raise_for_status()
   1434         if test_json:

/usr/local/dev/sar-pre-processing/env/lib/python3.9/site-packages/requests/models.py in raise_for_status(self)
    942         if http_error_msg:
--> 943             raise HTTPError(http_error_msg, response=self)
    944 

HTTPError: 500 Server Error: Internal Server Error for url: https://apihub.copernicus.eu/apihub/odata/v1/Products('82b56bfe-ae81-400d-9566-856ea5708c82')/$value

During handling of the above exception, another exception occurred:

ServerError                               Traceback (most recent call last)
<ipython-input-6-d92535fa5ff0> in <module>
      6     os.makedirs(path)
      7 except: FileExistsError
----> 8 api.download_all(products, path)

/usr/local/dev/sar-pre-processing/env/lib/python3.9/site-packages/sentinelsat/sentinel.py in download_all(self, products, directory_path, max_attempts, checksum, n_concurrent_dl, lta_retry_delay, **kwargs)
    721             info = self.get_product_odata(pid)
    722             product_infos[pid] = info
--> 723             is_online = not self.trigger_offline_retrieval(pid)
    724             if is_online:
    725                 online_prods[pid] = info

/usr/local/dev/sar-pre-processing/env/lib/python3.9/site-packages/sentinelsat/sentinel.py in trigger_offline_retrieval(self, uuid)
    643             self.logger.error(msg)
    644             raise ServerError(msg, r)
--> 645         _check_scihub_response(r, test_json=False)
    646 
    647     def download_all(

/usr/local/dev/sar-pre-processing/env/lib/python3.9/site-packages/sentinelsat/sentinel.py in _check_scihub_response(response, test_json, query_string)
   1477         elif 500 <= response.status_code < 600 or msg:
   1478             # 5xx: Server Error
-> 1479             raise ServerError(msg, response)
   1480         else:
   1481             raise SentinelAPIError(msg, response)

ServerError: HTTP status 500 Internal Server Error: UnsupportedOperationException : Error creating stream from file /mnt/s1b/dhus/ntc/incoming/x3/x3/x1/S1B_IW_SLC__1SDV_20210603T165832_20210603T165900_027195_033F94_5E37.zip
```

**Comment by TonioF** [comment]
The error message refers to something that is wrong server side. As you say you get a progress bar, can you please check whether the data is downloaded anyway? I reproduced your problem and found that despite the error messages everything could be downloaded.

**Comment by arthur-e** [comment]
Is the download asynchronous, then? Because I look at the Jupyter Kernel status and it is idle (i.e., the `ServerError` seems to be issue right before the program exits).

According to [a previous report about this error message](https://stackoverflow.com/questions/67582137/http-status-500-internal-server-error-unsupportedoperationexception-error-cr), the issue is intermittent, but I haven't yet found a time where it succeeds.

**Comment by TonioF** [comment]
No, the datasets are downloaded successively. To be clear, I get the same error - even multiple times - but in the end the progress bar is moving and it is downloaded. You get the error after the progress bar appears?

**Comment by arthur-e** [comment]
Yes, I see the error message after the progress bar appears.

**Comment by arthur-e** [comment]
To be more specific, the progress bar reads:

```
Fetching archival status: 0% [--------Progress bar-----------] 0/4 [00:00<?, ? products/s]
```

Where `[--Progress bar--]` stands in for the graphical display of an (empty) progress bar.

**Comment by TonioF** [comment]
This error could come from problems with the sentinelsats versions. Which version of sentinelsat are qou using?

Also, can you confirm that the 'products' that are the result of the api.query()-call is an OrderedDict?

**Comment by arthur-e** [comment]
I'm using version 1.0.1 of `sentinelsat`. It was installed automatically as part of the `sar-pre-processing` installation. And, yes, the result of the `api.query()` call, `products` is an instance of `collections.OrderedDict`.

**Comment by TonioF** [comment]
I'm still struggling to reproduce this error. My best guesses are to try again or use this bit instead of api.download_all(products, path):

for product in products.keys():
    api.download(product, path)

From what I could gather this is a problem that some users experience with sentinelsat / copernicus hub , and there are times when the request could be executed later. Anyway, could you try with the snippet above?

**Comment by McWhity** [comment]
There is now a new version of sentinelsat available (1.1.0). This should hopefully fix the problem. I included in the [jupyter notebook](https://github.com/multiply-org/sar-pre-processing/blob/master/docs/notebooks/default_process_single_image.ipynb) some more information how to download Sentinel-1 data. You can download Sentinel-1 data from the Copernicus Data Hub manually or via the new version of the sentinelsat package. If the Copernicus Data Hub is not working an alternative is provided with the Alaska Satellite Facility.

**Comment by McWhity** [comment]
I think the new version of sentinesat and the provision of an alternative (Alaska Satellite Facility) source has solved this issue. If there are no objections, I would like to close this issue.
